design/specs/table-import.md

```diff
@@ -14,153 +14,175 @@ Adding a new table from a file import allows users to input data into Mathesar w
 While data file import is a baseline feature for most applications in this category, additional functionality will be implemented in the future to enhance the data import process, such as automatic type detection, data cleansing, error detection, and others.
 
 # User Experience
-## Impact
-This feature will improve the user experience for users who need to input data into Mathesar by reducing the required effort when populating tables.
 
-This feature will also support the experience of first-time users, as it is expected by users of similar applications and would cause dissatisfaction if missing.
+## Navigating to the Data Import page
 
-## User Flow
+![](https://share.balsamiq.com/c/p3dqSXr66qa6EnPyykqpyZ.png)
 
-![](/assets/design/specs/table-import/Z3ngzR9.png)
+Data import navigation link will be found on the active schema page. Since we are only allowing users to import tables for now, positioning it with the existing tables/ active schema view made sense. 
 
+## Importing data into a new table
 
-### User Adds a Table
-The user starts the process of adding a table by clicking on any of the available options. The user can add a table from the sidebar menu(1) or the 'New' menu located on the top navigation bar in the main content area.
+![](https://share.balsamiq.com/c/jyszUUBfSvoCqd6qWQcz3G.png)
 
-![](/assets/design/specs/table-import/DH3XRuj.png =400x)
+### Defining the New Table
 
-#### Notes
-- We can implement the 'New' menu at a later iteration.
+When the user jumps to the upload view. They see a name input which can be added or not added. If not added by user the table will be given a default name and that can also be changed after the import is confirmed.
 
-### User Sets Table Options
-Once the add table process starts, the user is prompted to set a name, confirm, or select the schema that will hold the table.
-Using the lookup field, the user could also add a new schema from this dialog.
+### Uploading the Data File
 
-![](/assets/design/specs/table-import/ByfztC0tu.png =400x)
+#### File Upload Methods
 
-#### Valid and Invalid Table Names
-Based on [PostgreSQL Documentation](https://www.postgresql.org/docs/13/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS), some naming conventions apply to objects, including tables, views, and columns. For most cases, we'll handle any necessary quoting automatically and transparently and generate unique names when required by adding a number as the last character.
+There are three main blocks that specifies three ways to upload the file/data for import. Each block has some important description with it. Each active way is highlighted when in use.
```

Comment:
I think I disagree here. Tabs are also used for navigation. Plus here the user is accomplishing the same task but has different choices. And tabs are one of the most scannable components out there. So I am not sure how users are gonna miss them.

The current UI that is the radio buttons are bit more confusing since its a type of an input and not something user would perceive as a navigating tool.

Reply:
> Plus here the user is accomplishing the same task but has different choices. 

This is my point exactly, the user has to make a single choice of import method here and we need to make it clear that they have three choices and they can pick one. I don't think we need to use the radio buttons (although they do fulfill that purpose) but we need something that shows all of the options in a visually prominent way. With vertical tabs, it's easy for people to focus on the center of the page and miss that there are other options on the side.

Reply:
<img width="1772" alt="temp" src="https://user-images.githubusercontent.com/34674462/186217146-36209f5f-cf00-4d76-a40c-1fe6bcc44031.png">
I am not sure how vertical tabs are not visible enough. This is one example of tabs used for import file itself. We can I guess play around with the size of tabs and move the tabs inside the rectangle so they are visible enough?

Reply:
Those are more visible and a design like that would resolve my concerns about visibility. It still comes across as a navigation pattern, and not a pattern related to the the user needing to pick a single option out of a few available choices.